core_backend/app/question_answer/utils.py

```diff
@@ -26,6 +26,10 @@ def get_context_string_from_retrieved_contents(
     """
     context_list = []
     for i, result in content_response.items():
-        context_list.append(f"{i+1}. {result.retrieved_title}\n{result.retrieved_text}")
+        if not isinstance(result, QuerySearchResult):
+            result = QuerySearchResult(**result)
+        context_list.append(
+            f"{int(i)+1}. {result.retrieved_title}\n{result.retrieved_text}"
+        )
```

Comment:
This is useful for evaluation when I do /embeddings-search first, and use that response to test different RAG functions (prompts/rails)